packages/core/utils/src/types.ts

```diff
@@ -1,7 +1,7 @@
 import type * as Koa from 'koa';
 import type {} from 'koa-body';
 
-type ID = number | `${number}`;
+type ID = number | string;
```

Comment:
not sure how to handle this case! strapi/utils is imported from strapi/types, so I couldnt import the strapi types entity id here

Reply:
Why are we using strapi/utils from strapi/types? imo strapi/types should be a standalone :thinking: 
Seems odd but maybe there is a valid reason?

Reply:
Apparently we are importing the [providerFactory typing definitions ](https://github.com/strapi/strapi/blob/fc86e60e003758b414a1879364d99b1cab94289f/packages/core/types/src/modules/content-api.ts)from there, what do you think if we move those type definitions into strapi types too?

Reply:
Meh... The provider factory types does belong in the utils.
I think it would make more sense for the ContentAPI types to be located in strapi/strapi instead. But that's a bigger refacto. Let's keep the duplicated definition here